### PR TITLE
util: Rewrite makePath

### PR DESCRIPTION
**Pull request by NiLuJe**
The previous iteration, besides failing to handle leaf-only input, was relying on splitFilePathName, which just doesn't do what's required to incrementally build the directory tree the right way around ;).

This should more closely match mkdir -p, i.e., it will *fail* if any part (or all of it) of the path exists but is *not* a directory.

Re #10074

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/10111)

### CODE REVIEW

File: frontend/util.lua [hunk 1]
```diff
@@ -763,15 +763,33 @@ end
 -- @string path the directory to create
 -- @treturn bool true on success; nil, err_message on error
 function util.makePath(path)
-    path = path:gsub("/+$", "")
-    if util.pathExists(path) then return true end
+    if lfs.attributes(path, "mode") == "directory" then
+        return true
+    end
```
Comment (Frenzie) [inline frontend/util.lua:768]: Hm, what if it's a file?

Comment (NiLuJe) [reply frontend/util.lua:768]: It'll go through and fail via mkdir ;).

Comment (NiLuJe) [reply frontend/util.lua:768]: Specifically:

```
> print(makePath("baz"))
nil     File exists (creating `baz/` for `baz`)
> print(makePath("baz/"))
nil     File exists (creating `baz/` for `baz/`)
```

Comment (Frenzie) [reply frontend/util.lua:768]: Feels potentially a bit late/unclear if we already know right there that if ~= "directory" and ~= nil then danger.

Comment (NiLuJe) [reply frontend/util.lua:768]: That saves us from writing a dedicated early trap, and allows it to work no matter where the spurious file is in the tree (e.g., foo/bar/baz if bar is the file).

Comment (NiLuJe) [reply frontend/util.lua:768]: It also automagically handles the "trailing slash or not" conundrum, as `foo` == `foo/` for `mkdir`, but not necessarily for `stat`:

```
> print(lfs.attributes("baz", "mode"))
file
> print(lfs.attributes("baz/", "mode"))
nil     cannot obtain information from file 'baz/': Not a directory     20
```

**Pull request merged**